### DISCUSSION TITLE
Running locally with Docker Compose without using conda on Windows 11 OS

### DISCUSSION THREAD

**Discussion by asemandeveloper** [langgenius/dify#4848] [General]
Hello to all
How can I run the project with Docker Compose without using the docker hub images uploaded by langgenius? Actually, I want to use the modified source code in the project that is on my computer.
Guide me. 
Thanks a lot

**Comment by patryk20120** [comment]
You can:
1. Build images for `api` and `web` container from your local files with `docker build` and then use custom images in docker-compose files
2. Run project from source files https://docs.dify.ai/getting-started/install-self-hosted/local-source-code

**Reply by asemandeveloper** [reply]
Thank you Patryk
I am new to your nice project.
Please guide step by step.
How to use custom images in docker-compose files?

**Reply by patryk20120** [reply]
You have to change `image` value in your yaml file

**Comment by almassolarenrgi** [comment]
[]()
